.github/workflows/deploy.yml

```diff
@@ -4,34 +4,29 @@ on:
     tags:
       - '*'
 
+permissions:
+  contents: write
+      
 jobs:
   deploy:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Setup JDK
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v3
         with:
           distribution: 'temurin'
           java-version: 11
       - name: Build distribution
         run: ./gradlew :adapter:distZip
       - name: Create release
-        uses: actions/create-release@v1
-        id: create_release
+        uses: softprops/action-gh-release@v1
```

Comment:
I would probably just use the `gh` CLI like we do for the language server, that way we don't have to pull in another third-party action: https://github.com/fwcd/kotlin-language-server/blob/eee8afa9b6b92c4c5ae445823b3719818d595ec1/.github/workflows/deploy.yml#L27-L45